### PR TITLE
First-Timer Issue #201

### PR DESCRIPTION
**Pull request by Alcofribas**
Tried to refactor all instances of debug console output by introducing a private function `logMessage()` as requested in Issue #201. Thank you for offering this newbie task and providing meticulous directions – just the right thing for the very first pull request  :clap:

## Description
As described in Issue #201. 

## Motivation and Context
Minor code revision to eliminate some redundancy concerning the styling of the emphasized part in the logging string, setting `debugStyle` as the default style.

## How Has This Been Tested?
Succesfully tested by running `grunt --force`. The test specs  `test/spec/BasicSpec.js` have been adapted in line 755 in order to match a change in the main script made necessary by the refactoring. I suspect that I should have added some tests concerning the new function, but this was slightly beyond my present testing skills… 

## Types of changes

- [x] Code Refactoring (no change of external behavior intended) :pray: 
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have modified tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,5 +38,8 @@
     "grunt-template-jasmine-istanbul": "^0.4.0",
     "grunt-template-jasmine-requirejs": "^0.2.3",
     "grunt-markdox": "^1.2.1"
+  },
```
Comment (Alcofribas) [inline package.json]: It looked as if this was required, but maybe I just got something wrong?

**Comment by TalAter** [comment]
Hi @Alcofribas, I am sorry but I have already merged a part of the contribution to this from another user.

I hope you at least enjoyed the exercise and the chance to get started on GitHub, and you will do it some more in the future. I will try and create more first timer only issues to this end.

I have gone ahead and added some comments below, in the hopes that you may benefit from them.

Thank you again for your contribution :octocat: 👍

File: annyang.js [hunk 2]
```diff
@@ -113,9 +113,9 @@
         if (result) {
           var parameters = result.slice(1);
           if (debugState) {
-            console.log('command matched: %c'+currentCommand.originalPhrase, debugStyle);
+            logMessage('command matched: %c'+currentCommand.originalPhrase);
             if (parameters.length) {
-              console.log('with parameters', parameters);
+              logMessage('with parameters "'+parameters.join('", "')+'"');
```
Comment (TalAter) [inline annyang.js]: This changes the API. What was previously logged a certain way, is now logged differently.

I would personally think many times before doing a breaking change like that. It may be an edge case, but to some users, it may break their code (or at the very least, their tests).

File: annyang.js [hunk 3]
```diff
@@ -128,6 +128,15 @@
     invokeCallbacks(callbacks.resultNoMatch, results);
   };
 
+  var logMessage = function(logMsg, emphStyle) {
+    emphStyle = emphStyle || debugStyle;
+    if (/%c/.test(logMsg)) {
```
Comment (TalAter) [inline annyang.js]: Why this check? Won't this mean we can no longer pass a second argument to logMessage (like we did in line 118 before the change)

File: package.json (hunk 1 above)
Comment (TalAter) [inline package.json]: The build should pass even without this.
Proof: https://travis-ci.org/TalAter/annyang/builds/132315158

File: annyang.js (hunk 2 above)
Comment (Alcofribas) [inline annyang.js]: Good point, of course, it was indeed one of two details that didn't feel entirely right to myself, at least not very elegant. The other was the second one you commented on.

File: annyang.js (hunk 3 above)
Comment (Alcofribas) [inline annyang.js]: Honestly: This is just crap, and what makes it obvious is that you have to change line 118 to make it actually work. Things that happen during nighttime when motivation and mental performance are flagrantly drifting apart …

**Comment by Alcofribas** [comment]
Hi @TalAter, many thanks for your comments. There's a lot to learn for me from this, and hey – it was fun! I will go on and hopefully fail better, next time. I know that providing first timer issues may often cause some extra work instead of saving you time and effort, so I really appreciate you taking the trouble! See you soon on Github :+1:
